cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -828,14 +830,24 @@ KafkaAutoRebalancingReconciler kafkaAutoRebalancingReconciler() {
         }
 
         /**
-         * Run the reconciliation pipeline for Kafka auto-rebalancing
+         * Based on the autorebalance being enabled or not within Cruise Control configuration:
+         * - run the reconciliation pipeline for Kafka auto-rebalancing ...
+         * - ... create the auto-rebalance status on Idle if the cluster is still in the creation phase (no auto-rebalancing to run yet) or ...
+         * - ... reset to null the auto-rebalance status because the auto-rebalance was disabled
          *
          * @return  Future with Reconciliation State
          */
         Future<ReconciliationState> reconcileKafkaAutoRebalancing() {
-            return kafkaAutoRebalancingReconciler()
-                    .reconcile(kafkaStatus)
-                    .map(this);
+            if (kafkaAssembly.getSpec().getCruiseControl() == null || kafkaAssembly.getSpec().getCruiseControl().getAutoRebalance() == null) {
+                LOGGER.debugCr(reconciliation, "Cruise Control or inner autorebalance field not defined in the Kafka custom resource, no auto-rebalancing to reconcile");
+                // enforce no auto-rebalance status, if we are disabling Cruise Control and/or auto-rebalance from its configuration
+                kafkaStatus.setAutoRebalance(null);
+                return Future.succeededFuture(this);
+            } else {
+                return kafkaAutoRebalancingReconciler()
+                        .reconcile(kafkaStatus)
+                        .map(this);
+            }
```

Comment:
Why not turn this around?

```suggestion
            if (isAutoRebalancingEnabled()) {
                return kafkaAutoRebalancingReconciler()
                        .reconcile(kafkaStatus)
                        .map(this);
            } else {
                LOGGER.debugCr(reconciliation, "Cruise Control or inner autorebalance field not defined in the Kafka custom resource, no auto-rebalancing to reconcile");
                // enforce no auto-rebalance status, if we are disabling Cruise Control and/or auto-rebalance from its configuration
                kafkaStatus.setAutoRebalance(null);
                return Future.succeededFuture(this);
            }
```

Wouldn't that be more readable and reuse the existing method?

Reply:
Yeah, makes much more sense.